keel-core/src/main/kotlin/com/netflix/spinnaker/keel/actuation/EnvironmentPromotionChecker.kt

```diff
@@ -179,6 +188,50 @@ class EnvironmentPromotionChecker(
     }
   }
 
```

Comment:
Utility functions to make sure it's easy to check constraints in the right order (implicit comes before user defined)